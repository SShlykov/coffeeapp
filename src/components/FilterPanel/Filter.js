import React, {Component} from "react";

export default class Filter extends Component {
    state = {
        buttons: [
            {name: "Brazil"},
            {name: "Kenya"},
            {name: "Columbia"},
        ]
    }
    render() {
        const {filter, onFilterChange} = this.props;
        const buttons = this.state.buttons.map(({name}) => {
            const isActive = filter === name;
            const clazzName = isActive? "shop__filter-btn active": "shop__filter-btn";
            return(
                <button
                    key={name}
                    className={clazzName}
                    onClick={()=> onFilterChange(name)}
                >
                    {name}
                </button>
            )
        });
        return(
            <div className="shop__filter">
                <div className="shop__filter-label">
                    Or filter
                </div>
                <div className="shop__filter-group">
                    {buttons}
                </div>
            </div>
        )
    }
}