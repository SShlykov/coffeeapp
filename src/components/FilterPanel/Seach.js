import React, {Component} from "react";

export default class Search extends Component {
    state = {
        term: ""
    };
    onType = e => {
        const term = e.target.value;
        this.setState({term});
        this.props.onSearchChange(term);
    };
    render() {
        return (
            <>
                <label className="shop__search-label" htmlFor="filter">Looking for</label>
                <input
                    id="filter"
                    type="text"
                    placeholder="start typing here..."
                    className="shop__search-input"
                    value={this.state.term}
                    onChange={this.onType}
                />
            </>
        )
    }
}