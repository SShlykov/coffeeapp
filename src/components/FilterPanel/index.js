import FilterOrSearchPanel from "./FilterPanel";
export default FilterOrSearchPanel;
