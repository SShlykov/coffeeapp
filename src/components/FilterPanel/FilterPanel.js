import React from "react";
import Search from "./Seach";
import Filter from "./Filter";

const FilterPanel = ({onSearchChange, onFilterChange, filter}) => {
        return(
            <>
                <div className="row">
                    <div className="col-lg-4 offset-lg-2">
                        <Search onSearchChange={onSearchChange}/>
                    </div>
                    <div className="col-lg-4">
                        <Filter onFilterChange={onFilterChange} filter={filter}/>
                    </div>
                </div>
            </>
        )
};
export default FilterPanel;