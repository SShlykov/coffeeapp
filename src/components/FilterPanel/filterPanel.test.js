import React from 'react';
import FilterPanel from "./FilterPanel";
import {shallow} from 'enzyme';

describe('Testing <FilterPanel/>', () => {
    const panel = shallow(<FilterPanel/>);
    describe('Testing snap & state', () => {
        it('FilterPanel bar have rendered correctly', () => {
            expect(panel).toMatchSnapshot();
        });    
    });
});