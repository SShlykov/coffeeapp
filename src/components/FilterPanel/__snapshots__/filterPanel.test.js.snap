// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <FilterPanel/> Testing snap & state FilterPanel bar have rendered correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FilterPanel />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        className="row"
      >
        <div
          className="col-lg-4 offset-lg-2"
        >
          <Search />
        </div>
        <div
          className="col-lg-4"
        >
          <Filter />
        </div>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="col-lg-4 offset-lg-2"
          >
            <Search />
          </div>,
          <div
            className="col-lg-4"
          >
            <Filter />
          </div>,
        ],
        "className": "row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Search />,
            "className": "col-lg-4 offset-lg-2",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onSearchChange": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Filter />,
            "className": "col-lg-4",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "filter": undefined,
              "onFilterChange": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="row"
        >
          <div
            className="col-lg-4 offset-lg-2"
          >
            <Search />
          </div>
          <div
            className="col-lg-4"
          >
            <Filter />
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="col-lg-4 offset-lg-2"
            >
              <Search />
            </div>,
            <div
              className="col-lg-4"
            >
              <Filter />
            </div>,
          ],
          "className": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Search />,
              "className": "col-lg-4 offset-lg-2",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "onSearchChange": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Filter />,
              "className": "col-lg-4",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "filter": undefined,
                "onFilterChange": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
